monai/transforms/utility/array.py

```diff
@@ -1582,8 +1587,8 @@ def __call__(self, img: NdarrayOrTensor, meta_dict: dict | None = None) -> Ndarr
             self.filter = ApplyFilter(self.filter)
 
         img_ = self._apply_filter(img_)
-        if meta_dict:
-            img_ = MetaTensor(img_, meta=meta_dict)
+        if meta_dict or applied_operations is not None:
```

Comment:
```suggestion
        if meta_dict is not None or applied_operations is not None:
```
I think this check should have always been against None.